packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import { deleteUser, translate, getLogger } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = getLogger('Auth');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('DELETING');
+    try {
+      if (handleDelete) {
+        /*
+         * run custom delete handler, if provided. We pass `user` so that
+         * developer can do whichever cleanup with the user object they wish.
+         */
+        await handleDelete(user);
+      } else {
+        // else, run default deleteUser function.
+        await deleteUser();
+      }
+      setState('DONE');
+      onSuccess?.();
+    } catch (e) {
+      const error = e as Error;
+      setState('IDLE');
+      setErrorMessage(error.message);
+      onError?.(error);
+    }
+  }, [handleDelete, onError, onSuccess, user]);
+
+  // called when end user cancels account deletion confirmation
+  const handleCancel = React.useCallback(() => {
+    setState('IDLE');
+  }, []);
+
+  const handleConfirmDelete = React.useCallback(() => {
+    runDeleteUser();
+  }, [runDeleteUser]);
+
+  // Return null if Auth.getCurrentAuthenticatedUser is still in progress
+  if (isLoading) {
+    return null;
+  }
+
+  /** Return null if user isn't authenticated in the first place */
+  if (!user) {
+    logger.warn('<DeleteUser /> requires user to be authenticated.');
+    return null;
+  }
```

Comment:
Kind of just a thought (really more of a code style thing) and totally not required, but think it can be useful to order early return conditionals by whether a conditional excludes the next from running. Like, if `isLoading` is `true` it doesn't matter if `user` is falsy, same if state is `Done`, so the order of these conditionals would read like:
```
if (!user) {} // place before `DONE` check since it logs an error
if (state === 'DONE') {}
if (isLoading) {}
return ()

Reply:
Done in [c03a451](https://github.com/aws-amplify/amplify-ui/pull/2927/commits/c03a45186d872036e4955a86ebf9f43f5b21e397)!